layers/+spacemacs/spacemacs-editing/packages.el

```diff
@@ -539,7 +539,13 @@
             ;; Emacs GC which truncages the undo history very aggresively
             undo-limit 800000
             undo-strong-limit 12000000
-            undo-outer-limit 120000000)
+            undo-outer-limit 120000000
+            undo-tree-history-directory-alist
+            `(("." . ,(expand-file-name
+                       (let ((dir (concat spacemacs-cache-directory "undo-tree-history")))
+                         (unless (file-exists-p dir)
+                           (make-directory dir))
+                         dir)))))
```

Comment:
The logic here will fail if there exists a *file* in the cache directory titled `undo-tree-history`. I employed the following in my own fork:
```
undo-tree-history-directory-alist (list
                                   (cons
                                    "." (file-name-as-directory
                                         (concat spacemacs-cache-directory
                                                 "undo-tree")))))

```
`undo-tree` ensures that the directory exists so you don't have to worry that it has yet to be created.

Reply:
Thanks! It's really an issue here.

Would it be better to generate a warning when the path does exist but not a directory? I use function `file-accessible-directory-p` to achieve this in my new commit.